#!/usr/bin/env bash
set -euo pipefail

# Supondo que se tenha a seguinte estrutura no diretório de trabalho (work_dir/)

# work_dir/
# |
# |-- Genomes/
# |
# |-- Spades/
# |  |
# |  |--Sample1/contigs.fasta
# |  |
# |  |--Sample2/contigs.fasta

# O perl script fastagrep.pl pode ser obtido em https://github.com/rec3141/rec-genome-tools/blob/master/bin/fastagrep.pl

# (filter contigs – Generate final genomes in Genomes\ dir without contigs with COVERAGE < 2 OR SIZE < 500)

for i in `ls -1 SPAdes/*/contigs.fasta | sed 's/SPAdes\///' | sed 's/\/contigs\.fasta//'`

do

grep -F ">" SPAdes/${i}/contigs.fasta | sed -e 's/_/ /g' | sort -nrk 6 | awk '$6>=2.0 && $4>=500 {print $0}' | sed -e 's/ /_/g'| sed -e 's/>//g' > SPAdes/${i}/HCov.contigs.list

grep -F ">" SPAdes/${i}/contigs.fasta | sed -e 's/_/ /g' | sort -nrk 6 | awk '$6<2.0 || $4<500 {print $0}'  | sed -e 's/ /_/g'| sed -e 's/>//g' > SPAdes/${i}/LCov.contigs.list

perl fastagrep.pl -f SPAdes/${i}/HCov.contigs.list SPAdes/${i}/contigs.fasta > Genomes/${i}.fna

done


# Agora listo abaixo os passos metodológicos do projeto da Camila:

# Genome annotation with Prokka version 1.14.6

prokka --quiet --cpus 10 --compliant --kingdom Bacteria --genus Mycobacterium --outdir results/PROKKA/SAMPLE/ /data/Final_dataset/SAMPLE.fna > results/PROKKA/ SAMPLE.log


# Orthology analysis with OrthoFinder version 2.5.2


# * Before, you need create Orthofinder/DB folder with .faa files generated by PROKKA. You can just create symbolic links into Orthofinder/DB folder refering .faa files generated in results/PROKKA/SAMPLE/*.faa


nohup orthofinder -f results/Orthofinder/DB/ -t 40 -a 40 -y -M msa -n MYCs -o results/Orthofinder/Output > results/Orthofinder/orthofinder.log &


# Phylogeny - RAxML-NG version 1.0.1

nohup raxml-ng --threads 40 --all --bs-trees 500 --seed 1234 --model LG+G+F --msa ../Orthofinder/Output/MultipleSequenceAlignments/SpeciesTreeAlignment.fa --prefix MYCs > /dev/null 2>&1 &

# The resultant tree file with bootstraps and branch lengths will be named as MYCs.raxml.support

# After these steps, you can import the file MYCs.raxml.support into iTol tool (https://itol.embl.de). My configuration options to plot the final tree are: (1: Basic – Mode – Rectangular; 2: Advanced – Midpoint root).
